qiskit/chemistry/qubit_transformations/fermionic_transformation.py

```diff
@@ -0,0 +1,518 @@
+                                               'invalid length {} should be {}'.
+                                               format(len(self._z2symmetry_reduction),
+                                                      len(z2_symmetries.symmetries)))
+                valid = np.all(np.isin(self._z2symmetry_reduction, [-1, 1]))
+                if not valid:
+                    raise QiskitChemistryError('z2symmetry_reduction tapering values list must '
+                                               'contain -1\'s and/or 1\'s only was {}'.
+                                               format(self._z2symmetry_reduction,))
+                z2_symmetries.tapering_values = self._z2symmetry_reduction
+
+            logger.debug('Apply symmetry with tapering values %s', z2_symmetries.tapering_values)
+            chop_to = 0.00000001  # Use same threshold as qubit mapping to chop tapered operator
+            z2_qubit_op = z2_symmetries.taper(qubit_op).chop(chop_to)
+            z2_aux_ops = []
+            for aux_op in aux_ops:
+                z2_aux_ops.append(z2_symmetries.taper(aux_op).chop(chop_to) if aux_op is not None
+                                  else None)
+
+        return z2_qubit_op, z2_aux_ops, z2_symmetries
+
+    @staticmethod
+    def _check_commutes(cliffords: List[WeightedPauliOperator],
+                        operator: WeightedPauliOperator) -> bool:
+        """
+        Check commutations
+
+        Args:
+            cliffords : cliffords
+            operator: qubit operator
+
+        Returns:
+            Boolean: does_commute
+        """
+        commutes = []
+        for clifford in cliffords:
+            commutes.append(operator.commute_with(clifford))
+        does_commute = np.all(commutes)
+        logger.debug('  \'%s\' commutes: %s, %s', operator.name, does_commute, commutes)
+        return does_commute
+
+    @staticmethod
+    def _pick_sector(z2_symmetries: Z2Symmetries, hf_str: np.ndarray) -> Z2Symmetries:
+        """
+        Based on Hartree-Fock bit string and found symmetries to determine the sector.
+        The input z2 symmetries will be mutated with the determined tapering values.
+
+        Args:
+            z2_symmetries: the z2 symmetries object.
+            hf_str: Hartree-Fock bit string (the last index is for qubit 0).
+
+        Returns:
+            the original z2 symmetries filled with the correct tapering values.
+        """
+        # Finding all the symmetries using the find_Z2_symmetries:
+        taper_coef = []
+        for sym in z2_symmetries.symmetries:
+            # pylint: disable=no-member
+            coef = -1 if np.logical_xor.reduce(np.logical_and(sym.z[::-1], hf_str)) else 1
+            taper_coef.append(coef)
+        z2_symmetries.tapering_values = taper_coef
+        return z2_symmetries
+
+    def interpret(self, eigenstate_result: EigenstateResult) -> ElectronicStructureResult:
```

Comment:
I am not sure why this needs to interpret an eigenstate result rather than a min eig solver result. See prior comments in the MESGndStatCalc

Reply:
See reply to prior comment.

Reply:
I think this leads to difficulties with more directly using a result - maybe it should be a Union of this and (Min)EigResult to allow a end user to take the operators, user VQE and then pass the result directly in. It seems that the 'conversion' to some EigenstateResult should be simple enough here.

Reply:
Let's keep the discussion here: https://github.com/Qiskit/qiskit-aqua/pull/1288#discussion_r502533896